Common/include/basic_types/ad_structure.hpp

```diff
@@ -66,7 +66,7 @@ inline void PrintStatistics() {}
  * graph. \param[in] data - The variable to be registered as input. \param[in] push_index - boolean whether we also want
  * to push the index.
```

Comment:
Nice stuff :+1: is there any way to get warned about this in the future?
Also looks like we can update this comment.

Reply:
Thanks for noting, I updated the comment. `AD::SetDerivative` used to have an implicit size check and automatic resizing if needed. This was removed in #2039 because it harms the hybrid AD performance if called in parallel, like from the `InitializeAdjoint` functions. I will add assertions to CoDiPack though, so that there are warnings about this when debugging with `-DCODI_EnableAssert`.

Reply:
Makes sense, I updated CoDiPack and added the flag